src/mot/tracker/tracker.py

```diff
@@ -1,47 +1,169 @@
-        return cv2.warpAffine(im, matrix, self.img_shape, flags=cv2.INTER_LINEAR + cv2.WARP_INVERSE_MAP)
-
-    def warp_coords(self, coords, matrix):
+import copy
+
+from mot.object_detection.utils import np_box_ops
+from mot.object_detection.config import config as cfg
+
+
+class Trash():
+    '''Detected trash class
+    '''
+    def __init__(self, id, label, box, frame):
+        self.id = id
+        self.label = label
+        self.boxes = [box]
+        self.frames = [frame]
+
+    def add_matching_object(self, box, frame):
+        self.boxes.append(box)
+        self.frames.append(frame)
+
+    def get_np_box(self):
+        return np.array(self.boxes[-1])
+
+    def find_best_match_in_list(self, list_trash, iou_threshold):
+        '''Finds the best matching trash index with regards to a list of trash
+        matching is defined by same class and best iou between boxes
+
+        Arguments:
+        - list_trash: A list of reference trash
+        - iou_threshold: A float IoU threshold
+
+        Returns:
+        - None, or the index of the matching trash
         '''
-        Assure coords is a np array of shape (X, 2)
+        matching_id = None
+        trash_box = self.get_np_box()[np.newaxis,:]
+        boxes_list = []
+        idx_list = []
+        for matching_trash in list_trash:
+            if self.label == matching_trash.label:
+                boxes_list.append(matching_trash.get_np_box())
+                idx_list.append(matching_trash.id)
+
+        if boxes_list:
+            ious = np_box_ops.iou(trash_box, np.vstack(boxes_list))
+            best_iou_idx = np.argmax(ious,axis=-1)[0]
+            assert ious.shape == (1, len(boxes_list))
+            if ious[0,best_iou_idx] >= iou_threshold:
+                matching_id = idx_list[best_iou_idx]
+
+        return matching_id
+
+    def get_center(self):
+        x1,y1,x2,y2 = self.boxes[-1]
+        return (x2+x1)/2, (y2+y1)/2
+
+    def __repr__(self):
+        return "(id:{}, label:{}, center:({:.1f},{:.1f}), frames:{})".format(self.id, self.label, self.get_center()[0], self.get_center()[1], self.frames)
+
+    def json_result(self):
+        classname = cfg.DATA.CLASS_NAMES[self.label]
+        return {"label":classname, "frames":self.frames, "id":self.id}
+
+
+class ObjectTracking():
+    '''Wrapper class to tracking trash objects in video output frames
+    '''
+    def __init__(self, video_id, list_path_images, list_inference_output = None, fps = 2, list_geoloc = None):
+        self.video_id = video_id
+
+        self.list_path_images = list_path_images
+        self.list_inference_output = list_inference_output
+
+        self.num_images = len(list_path_images)
+        self.list_geoloc = list_geoloc
+        self.fps = fps
+
+        self.detected_trash = []
+        self.tracking_done = False
+
+        self.iou_threshold = 0.3
+        self.rewind_window_match = 2
+
+    def potential_matching_trash_list(self, frame_idx, objects_per_frame_list):
```

Comment:
Could rename  objects_per_frame_list to objects_per_frame.
Doctyping will make it clear that this is a list.
```python
def potential_matching_trash_list(self, frame_idx: int, objects_per_frame: list):
    pass
```